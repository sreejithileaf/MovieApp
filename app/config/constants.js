/**
 * Created on July 16, 2022
 * Constants - Constant data of the App.
 */

import { Dimensions, Platform } from "react-native";

export default constants = {
  APP_NAME: "MovieApp",
  APP_BASE_URL: "http://api.themoviedb.org/3/movie/",
  APP_IMG_URL_SMALL: "https://image.tmdb.org/t/p/w185",
  APP_IMG_URL_LARGE: "https://image.tmdb.org/t/p/w300",

  APP_BASE_HEIGHT: 896,
  APP_BASE_WIDTH: 414,

  Fonts: {
    REGULAR: Platform.OS === "ios" ? "Poppins-Regular" : "Poppins_Regular",

    BOLD: Platform.OS === "ios" ? "Poppins-Regular" : "Poppins_Bold",

    MEDIUM: Platform.OS === "ios" ? "Poppins-Regular" : "Poppins_Medium",
  },

  APP_THEME_COLOR: "rgb(251,125,29)",
  APP_WHITE_COLOR: "#FFFFFF",
  APP_LIGHTGREY_COLOR: "#EEEEEE",
  APP_GENRES_COLOR: "#3A3F47",
  APP_GREEN_COLOR: "#4FCCA3",
  APP_TAB_COLOR: "#242A32",
  APP_BLACK_COLOR: "rgb(59,59,59)",
  APP_DARK_BLACK_COLOR: "rgb(32,32,32)",
  APP_TEXT_GRAY_COLOR: "rgb(152,152,152)",

  ACTIVE_OPACITY: 0.7,

  IS_ANDROID: Platform.OS === "ios" ? false : true,

  IOS_VERSION: parseInt(Platform.Version, 10),
};
