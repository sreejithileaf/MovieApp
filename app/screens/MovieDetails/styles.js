/**
 * Created on July 17, 2022
 * Dashboard Screen - About Us Screen Styles
 */

import { StyleSheet } from "react-native";
import Constants from "../../config/constants";

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: Constants.APP_TAB_COLOR,
  },
  posterStyle: {
    height: 210,
    borderBottomLeftRadius: 16,
    borderBottomRightRadius: 16,
  },

  movieTitleContainer: {
    flexDirection: "column",
    marginStart: 20,
    marginEnd: 20,
  },

  imgStyle: {
    height: 125,
    width: 95,
    borderRadius: 15,
    position: "absolute",
    top: -60,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: "flex-start",
    alignItems: "flex-start",
  },

  movieTitleTextStyle: {
    fontSize: 18,
    color: Constants.APP_WHITE_COLOR,
    fontFamily: Constants.Fonts.BOLD,
    marginStart: 110,
    marginTop: 15,
  },

  movieTextStyle: {
    color: Constants.APP_WHITE_COLOR,
    fontFamily: Constants.Fonts.REGULAR,
    fontWeight: "500",
  },

  genreTextStyle: {
    alignItems: "flex-start",
    fontSize: 12,
    fontFamily: Constants.Fonts.MEDIUM,
    color: Constants.APP_LIGHTGREY_COLOR,
  },
  genreContainer: {
    marginTop: 15,
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 5,
    paddingBottom: 5,
    backgroundColor: Constants.APP_GENRES_COLOR,
    borderRadius: 16,
  },

  backButtonContainer: {
    flexDirection: "row",
    justifyContent: "center",
    width: 105,
    height: 42,
    backgroundColor: Constants.APP_GENRES_COLOR,
    borderRadius: 16,
  },

  watchListContainer: {
    justifyContent: "center",
    width: 42,
    height: 42,
    backgroundColor: Constants.APP_GREEN_COLOR,
    borderRadius: 16,
  },

  thinSeparator: {
    borderTopColor: Constants.APP_GENRES_COLOR,
    width: 90,
    borderTopWidth: 4,
    marginTop: 5,
  },
  genreListStyle: {
    flexGrow: 0,
    marginTop: 15,
  },
  aboutMovieStyle: {
    fontSize: 14,
    width: 90,
    marginTop: 15,
  },

  overviewStyle: {
    fontSize: 12,
    marginTop: 15,
  },

  overviewContentStyle: {
    fontSize: 12,
    marginTop: 5,
  },

  releaseStyle: {
    fontSize: 12,
    marginTop: 12,
  },
  releaseContentStyle: {
    fontSize: 12,
  },

  rowStyle: {
    flexDirection: "row",
  },
  avgRatingContainer: {
    flex: 1,
    flexDirection: "column",
    marginTop: 12,
    alignSelf: "flex-start",
  },

  bottomContainerStyle: {
    flexDirection: "row",
    marginHorizontal: 20,
    marginBottom: 10,
  },
  backButtonContainerStyle: {
    flex: 1,
    flexDirection: "column",
    alignSelf: "flex-start",
  },
  backButtonImageStyle: {
    width: 16,
    height: 14,
    alignSelf: "center",
    justifyContent: "center",
  },
  backTextStyle: {
    color: Constants.APP_LIGHTGREY_COLOR,
    marginStart: 10,
    fontSize: 14,
    alignSelf: "center",
  },
  watchListContainerStyle: {
    flex: 1,
    flexDirection: "column",
    alignItems: "flex-end",
  },
  watchListImageStyle: {
    width: 14,
    height: 18,
    alignSelf: "center",
    justifyContent: "center",
  },
});

export default styles;
