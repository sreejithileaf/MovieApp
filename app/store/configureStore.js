/**
 * Created on JUL 16, 2022
 * ConfigureStore - Store configuring.
 */

import sagas from "../sagas";
import reduxReset from "redux-reset";
import rootReducers from "../reducers";
import { createLogger } from "redux-logger";
import createSagaMiddleware from "redux-saga";
import { createStore, compose, applyMiddleware } from "redux";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { persistStore, persistCombineReducers } from "redux-persist";

const config = {
  key: "root",
  storage: AsyncStorage,
  whitelist: ["watchlistReducer"], //to persist reducer data
  blacklist: [], //to remove reducer to persist
  debug: true, //to get useful logging
};

const middleware = [];
const sagaMiddleware = createSagaMiddleware();

middleware.push(sagaMiddleware);

if (__DEV__) {
  middleware.push(createLogger());
}

const reducers = persistCombineReducers(config, rootReducers);
const enhancers = [applyMiddleware(...middleware), reduxReset()];

const persistConfig = { enhancers };
const store = createStore(reducers, undefined, compose(...enhancers));
const persistor = persistStore(store, persistConfig, () => {});
const configureStore = () => {
  return { persistor, store };
};

sagaMiddleware.run(sagas);

export default configureStore;
