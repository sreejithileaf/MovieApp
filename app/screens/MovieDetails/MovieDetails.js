/**
 * Created on July 17, 2022
 * Movie Details screen
 */

import {
  View,
  TouchableOpacity,
  Text,
  SafeAreaView,
  FlatList,
  Image,
  ToastAndroid,
} from "react-native";

import styles from "./styles";
import React, { Component } from "react";
import Images from "../../config/images";
import Constants from "../../config/constants";
import { ScrollView } from "react-native";
import HudView from "../../components/hudView";

export default class MovieDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      movieData: "",
      isWatchlistItem: false,
    };
  }

  /*
   * Fetch movie details on page load
   */

  componentDidMount = async () => {
    const { id } = await this.props.navigation.state.params;
    const { watchlistItems } = this.props;
    let filterData = watchlistItems.filter((item) => item.id == id);
    this.setState({ isWatchlistItem: filterData.length > 0 });
    this.props.getMovieDetailsAction(id, ({ data, status }) => {
      if (status) {
        console.log("MOVIE DETAIL SCREEN", data);
        this.setState({
          movieData: data,
        });
      }
    });
  };

  /*
   * Shows a toast/alert message when movie is added/removed from Watchlist
   */
  _notifyMessage(msg) {
    if (Platform.OS === "android") {
      ToastAndroid.show(msg, ToastAndroid.SHORT);
    } else {
      alert(msg);
    }
  }

  /*
   * Navigates to back
   */
  _didTapOnBackButton = () => {
    this.props.navigation.goBack();
  };

  /*
   * Add/Remove movie item to watchlist
   * Fetch watchlist data from reducer and modify it and dispatch to reducer
   */
  _addToWatchList = () => {
    const { item, id } = this.props.navigation.state.params;
    const { watchlistItems, updateWatchlist } = this.props;
    const { isWatchlistItem } = this.state;
    let updatedWatchListArray = [];
    if (isWatchlistItem) {
      let tempArray = [...watchlistItems];
      let indexFound = -1;
      tempArray.map((obj, index) => {
        if (obj.id == id) {
          indexFound = index;
        }
      });
      if (indexFound > -1) tempArray.splice(indexFound, 1);
      updatedWatchListArray = tempArray;
    } else {
      updatedWatchListArray = [...watchlistItems, ...[item]];
    }
    updateWatchlist(updatedWatchListArray);
    this.setState({ isWatchlistItem: !this.state.isWatchlistItem });
    this._notifyMessage(
      isWatchlistItem ? "Removed from Watchlist" : "Added to Watchlist"
    );
  };
  render() {
    const { movieData, isWatchlistItem } = this.state;
    const { isLoading } = this.props;

    return (
      <SafeAreaView style={styles.safeContainer}>
        <ScrollView showsVerticalScrollIndicator={false}>
          <Image
            style={styles.posterStyle}
            source={{
              uri: Constants.APP_IMG_URL_LARGE + movieData.poster_path,
            }}
            testID={"poster"}
          ></Image>

          <View style={styles.movieTitleContainer}>
            <Image
              style={styles.imgStyle}
              source={{
                uri: Constants.APP_IMG_URL_LARGE + movieData.poster_path,
              }}
              testID={"sub_poster"}
            ></Image>

            <Text style={styles.movieTitleTextStyle} testID={"title"}>
              {movieData.title}
            </Text>
          </View>

          <FlatList
            style={styles.genreListStyle}
            horizontal={true}
            data={movieData.genres}
            renderItem={({ item, key }) => {
              return (
                <View
                  style={[
                    {
                      marginLeft: key == 0 ? 20 : 10,
                      marginRight: key == movieData.genres.length - 1 ? 20 : 10,
                    },
                  ]}
                >
                  {!!item.name && (
                    <View style={styles.genreContainer}>
                      <Text style={styles.genreTextStyle}>{item.name}</Text>
                    </View>
                  )}
                </View>
              );
            }}
          />

          <View style={styles.movieTitleContainer}>
            <Text style={[styles.movieTextStyle, styles.aboutMovieStyle]}>
              {"About Movie"}
            </Text>

            <View style={styles.thinSeparator} />

            <Text style={[styles.movieTextStyle, styles.overviewStyle]}>
              {"Overviews:"}
            </Text>

            <Text style={[styles.movieTextStyle, styles.overviewContentStyle]}>
              {movieData.overview}
            </Text>

            <Text style={[styles.movieTextStyle, styles.releaseStyle]}>
              {"Release Date:"}
            </Text>

            <Text style={[styles.movieTextStyle, styles.releaseContentStyle]}>
              {movieData.release_date}
            </Text>

            <View style={styles.rowStyle}>
              <View style={styles.avgRatingContainer}>
                <Text
                  style={[styles.movieTextStyle, styles.releaseContentStyle]}
                >
                  {"Average Rating:"}
                </Text>

                <Text
                  style={[styles.movieTextStyle, styles.releaseContentStyle]}
                >
                  {movieData.vote_average}
                </Text>
              </View>

              <View
                style={{
                  flex: 1,
                  flexDirection: "column",
                  marginTop: 12,
                  alignItems: "flex-start",
                }}
              >
                <Text style={[styles.movieTextStyle, { fontSize: 12 }]}>
                  {"Rate Count:"}
                </Text>

                <Text style={[styles.movieTextStyle, { fontSize: 12 }]}>
                  {movieData.vote_count}
                </Text>
              </View>
            </View>

            <Text style={[styles.movieTextStyle, styles.releaseStyle]}>
              {"Popularity:"}
            </Text>

            <Text style={[styles.movieTextStyle, styles.releaseContentStyle]}>
              {movieData.popularity}
            </Text>
          </View>
        </ScrollView>

        <View style={styles.bottomContainerStyle}>
          <View style={styles.backButtonContainerStyle}>
            <TouchableOpacity
              style={styles.backButtonContainer}
              onPress={() => this._didTapOnBackButton()}
            >
              <Image
                style={styles.backButtonImageStyle}
                source={Images.back}
              ></Image>

              <Text style={[styles.movieTextStyle, styles.backTextStyle]}>
                {"Back"}
              </Text>
            </TouchableOpacity>
          </View>

          <View style={styles.watchListContainerStyle}>
            <TouchableOpacity
              style={styles.watchListContainer}
              onPress={() => this._addToWatchList()}
              testID={"watchlist"}
            >
              <Image
                style={styles.watchListImageStyle}
                source={
                  isWatchlistItem
                    ? Images.watchlist_dark_tick
                    : Images.watchlist_dark
                }
              ></Image>
            </TouchableOpacity>
          </View>
        </View>
        {isLoading && <HudView />}
      </SafeAreaView>
    );
  }
}
