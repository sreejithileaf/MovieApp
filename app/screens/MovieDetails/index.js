/**
 * Created on July 16, 2022
 * DashboardScreen - DashboardScreen container
 */

import { connect } from "react-redux";
import React, { Component } from "react";
import MovieDetails from "./MovieDetails";
import * as getMovieDetails from "../../actions/movieDetailsActions";
import * as watchListActions from "../../actions/watchlistActions";

class MovieDetailsScreenContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <MovieDetails {...this.props} />;
  }
}

function mapStateToProps(state) {
  return {
    isLoading: state.loadingReducer.isLoading,
    watchlistItems: state.watchlistReducer.watchlistItems,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getMovieDetailsAction: (params, getMovieDetailsCallback) => {
      dispatch(
        getMovieDetails.getMovieDetails(params, getMovieDetailsCallback)
      );
    },

    updateWatchlist: (watchlist) => {
      dispatch(watchListActions.updateWatchlist(watchlist));
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieDetailsScreenContainer);
