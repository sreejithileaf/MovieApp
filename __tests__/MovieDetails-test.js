/**
 * @format
 */

import "react-native";
import React from "react";
import MovieDetails from "../app/screens/MovieDetails/MovieDetails";
import constants from "../app/config/constants";
import { Image } from "react-native";
import { shallow, configure, mount } from "enzyme";
import { Provider } from "react-redux";
import store from "../app/store/configureStore";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { render, container, screen } from "@testing-library/react-native";
import { toHaveStyle } from "@testing-library/jest-native";
import "@testing-library/jest-native/extend-expect";

describe("Movie Details Screen ", () => {
  const wrapper = renderer.create(<MovieDetails />).toJSON();

  test("Snapshot ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Image must contain an src", async () => {
    const { getByAltText } = await render(<MovieDetails />);

    const image = screen.getByTestId("poster");
    expect(image.props.source.uri).toContain(
      "https://image.tmdb.org/t/p/w300undefined"
    );
  });
  it("Image must have a style", async () => {
    const { getByAltText } = await render(<MovieDetails />);

    const image = screen.getByTestId("sub_poster");
    expect(image).toHaveStyle({
      height: 125,
      width: 95,
      borderRadius: 15,
      position: "absolute",
      top: -60,
      left: 0,
      right: 0,
      bottom: 0,
      justifyContent: "flex-start",
      alignItems: "flex-start",
    });
  });
  it("It must have a about movie section", async () => {
    const { getByText } = render(<MovieDetails />);
    getByText("About Movie", { exact: false });
  });
  it("It must have a about Overviews section", async () => {
    const { getByText } = render(<MovieDetails />);
    getByText("Overviews", { exact: false });
  });
  it("It must have a Release Date", async () => {
    const { getByText } = render(<MovieDetails />);
    getByText("Release Date:", { exact: true });
  });
  it("It must have a about movie rating section", async () => {
    const { getByText } = render(<MovieDetails />);
    getByText("Average Rating:", { exact: false });
  });
});
