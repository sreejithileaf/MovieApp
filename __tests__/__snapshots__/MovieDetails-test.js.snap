// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Screen  Snapshot  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#242A32",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w300undefined",
          }
        }
        style={
          Object {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "height": 210,
          }
        }
        testID="poster"
      />
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginEnd": 20,
            "marginStart": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://image.tmdb.org/t/p/w300undefined",
            }
          }
          style={
            Object {
              "alignItems": "flex-start",
              "borderRadius": 15,
              "bottom": 0,
              "height": 125,
              "justifyContent": "flex-start",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": -60,
              "width": 95,
            }
          }
          testID="sub_poster"
        />
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Poppins-Regular",
              "fontSize": 18,
              "marginStart": 110,
              "marginTop": 15,
            }
          }
          testID="title"
        />
      </View>
      <RCTScrollView
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexGrow": 0,
            "marginTop": 15,
          }
        }
        viewabilityConfigCallbackPairs={Array []}
      >
        <View />
      </RCTScrollView>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginEnd": 20,
            "marginStart": 20,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 14,
                "marginTop": 15,
                "width": 90,
              },
            ]
          }
        >
          About Movie
        </Text>
        <View
          style={
            Object {
              "borderTopColor": "#3A3F47",
              "borderTopWidth": 4,
              "marginTop": 5,
              "width": 90,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
                "marginTop": 15,
              },
            ]
          }
        >
          Overviews:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
                "marginTop": 5,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
                "marginTop": 12,
              },
            ]
          }
        >
          Release Date:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "marginTop": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Poppins-Regular",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 12,
                  },
                ]
              }
            >
              Average Rating:
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Poppins-Regular",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 12,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "marginTop": 12,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Poppins-Regular",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 12,
                  },
                ]
              }
            >
              Rate Count:
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Poppins-Regular",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 12,
                  },
                ]
              }
            />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
                "marginTop": 12,
              },
            ]
          }
        >
          Popularity:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3A3F47",
            "borderRadius": 16,
            "flexDirection": "row",
            "height": 42,
            "justifyContent": "center",
            "opacity": 1,
            "width": 105,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/images/back.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 14,
              "justifyContent": "center",
              "width": 16,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Poppins-Regular",
                "fontWeight": "500",
              },
              Object {
                "alignSelf": "center",
                "color": "#EEEEEE",
                "fontSize": 14,
                "marginStart": 10,
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#4FCCA3",
            "borderRadius": 16,
            "height": 42,
            "justifyContent": "center",
            "opacity": 1,
            "width": 42,
          }
        }
        testID="watchlist"
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/images/watchlist_dark.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 18,
              "justifyContent": "center",
              "width": 14,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
